File: Neon.Web/Args/Client/InitializeArgs.cs
using Neon.Web.Models;

namespace Neon.Web.Args.Client;

public class InitializeArgs
{
    public required IReadOnlyCollection<ActiveUserModel> ActiveUsers { get; init; }
    public required IReadOnlyCollection<FriendModel> Friends { get; init; }
    public required IReadOnlyCollection<IncomingUserRequestModel> IncomingUserRequests { get; init; }
    public required IReadOnlyCollection<OutgoingUserRequestModel> OutgoingUserRequests { get; init; }
}

File: Neon.Web/wwwroot/ts/modules/args/initialize-args.ts
import { UserRequestTypeInt } from "../enums/user-request-type";

export default interface InitializeArgs {
    readonly activeUsers: ActiveUserModel[];
    readonly friends: FriendModel[];
    readonly incomingUserRequests: IncomingUserRequestModel[];
    readonly outgoingUserRequests: OtgoingUserRequestModel[];
}

interface ActiveUserModel {
    readonly key: string;
    readonly username: string;
}

interface FriendModel {
    readonly key: string;
    readonly username: string;
}

interface IncomingUserRequestModel {
    readonly type: UserRequestTypeInt;
    readonly requesterKey: string;
    readonly requesterUsername: string;
    readonly createdAt: Date;
}

interface OtgoingUserRequestModel {
    readonly type: UserRequestTypeInt;
    readonly responderKey: string;
    readonly responderUsername: string;
    readonly createdAt: Date;
}